rust_snuba/src/processors/release_health_metrics.rs

```diff
@@ -0,0 +1,754 @@
+use adler::Adler32;
+use anyhow::Context;
+use chrono::DateTime;
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+use serde::{Deserialize, Serialize};
+use std::{collections::BTreeMap, vec};
+
+use crate::{
+    types::{InsertBatch, RowData},
+    KafkaMessageMetadata, ProcessorConfig,
+};
+
+use super::utils::enforce_retention;
+
+const GRANULARITY_TEN_SECONDS: u8 = 0;
+const GRANULARITY_ONE_MINUTE: u8 = 1;
+const GRANULARITY_ONE_HOUR: u8 = 2;
+const GRANULARITY_ONE_DAY: u8 = 3;
+
+/// Generate a timeseries ID from the given parameters. Timeseries IDs are used to
+/// uniquely identify a timeseries in the database. This implemenation is based on
+/// the python implementation in order to ensure that the same timeseries ID is
+/// generated for the same input.
+///
+/// Upstream sends the tag keys and values as strings. Which is surprising since the
+/// tag keys are indexed so they should be integers. But we have to deal with it.
+fn generate_timeseries_id(
```

Comment:
Since there's quite a bit of overlap between generic metrics and metrics, how about putting these common functions into a shared metrics utils file and using it from both sets of processors?

Reply:
I think there are probably a few ways. You could have a custom trait `ToBytes` or something, and a u64 and string implementation of that trait.

But I assumed the 2 functions were identical before, and given Nikhar's comments about maintenance mode it's probably better to keep things separate as they are now.

Reply:
> You could have a custom trait ToBytes or something, and a u64 and string implementation of that trait.

How do I do that exactly? I'm assuming the trait would look something like this:

```rust
trait ToBytes {
    fn generate(&self) -> ???; // What do I do here?
}
```

If it is
```rust
impl ToBytes for String {
    fn generate(&self) -> [u8];
}
```
Then wouldn't it require me to know the size of that String at compile time?

Otherwise with
```rust
impl ToBytes for u32 {
    fn generate(&self) -> &[u8];
}
```
It would require me to borrow the [u8] from somewhere?

Reply:
I see. Making it `Vec<u8>` seems like the easiest way to me then. But I agree it doesn't seem worth the trouble in this specific scenario.

Reply:
Thanks!